app/assets/javascripts/shared/components/languageSwitcher.js.coffee

```diff
@@ -1,10 +1,8 @@
 angular.module('dahlia.components')
-.component 'languageSwitcher',
+.component 'languageSwitcher', {
   templateUrl: 'shared/components/language-switcher.html'
   controller: ['SharedService', '$state', '$scope', (SharedService, $state, $scope) ->
-    ctrl = @
-
-    @switchToLanguage = (lang) ->
+    $scope.stateForLanguage = (lang) ->
```

Comment:
@creativecoder this refactor makes me wonder about consolidating the welcome page language duties in angularInitialize. which would remove the need for this function entirely because then the language switcher would just always use the current state in its href. then angularInitialize, which is already parsing welcome page requests, would do the appropriate language-determining. but the one downside to that is that the language switcher would then be outputting technically incorrect paths for the welcome pages - e.g. `/es/welcome-chinese` - which would be correctly transformed by angularInitialize but still seems undesirable. so because of that i think how it is here is good.

Reply:
👍